internal/cli/common.go

```diff
@@ -0,0 +1,64 @@
+package cli
+
+import (
+	"errors"
+
+	"github.com/coreos/airlock/internal/config"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var (
+	// airlockCmd is the top-level cobra command for `airlock`
+	airlockCmd = &cobra.Command{
+		Use:               "airlock",
+		Short:             "Update/reboot manager, with distributed locking based on etcd3",
+		PersistentPreRunE: commonSetup,
+	}
+
+	configPath  string
+	runSettings *config.Settings
+)
+
+// Init initializes the CLI environment for airlock
+func Init() (*cobra.Command, error) {
+	airlockCmd.PersistentFlags().StringVarP(&configPath, "config", "c", "/etc/airlock/config.toml", "path to configuration file")
+
+	airlockCmd.AddCommand(cmdServe)
+
+	logrus.SetLevel(logrus.DebugLevel)
+	return airlockCmd, nil
+}
+
+func commonSetup(cmd *cobra.Command, cmdArgs []string) error {
+	var err error
+	if configPath == "" {
+		return errors.New("empty path to configuration file")
+	}
+
+	runSettings, err = config.Parse(configPath)
+	if err != nil {
+		return err
+	}
+
+	if err := validateSettings(*runSettings); err != nil {
+		return err
+	}
+
+	logrus.WithFields(logrus.Fields{
+		"endpoints": runSettings.EtcdEndpoints,
+	}).Debug("etcd3 configuration")
+
+	return nil
+}
+
+func validateSettings(cfg config.Settings) error {
+	if len(cfg.EtcdEndpoints) == 0 {
+		return errors.New("no etcd3 endpoints configured")
+	}
+	if len(cfg.LockGroups) == 0 {
```

Comment:
Can `LockGroups` ever actually have a length of 0? In `Parse` if it has a length of 0 the `default` key gets assigned to 1

Reply:
Not with the current logic, you are on spot. Though, I prefer to leave this sanity check in place to catch any possible future buggy rework around this.